modules/lti/src/main/java/org/opencastproject/lti/LtiUtils.java

```diff
@@ -0,0 +1,336 @@
+
+      Map<String, String> nextProp = new HashMap<String, String>();
+      // Convert to Map<String, String>
+      for (final Map.Entry<String, String> entry : params) {
+        nextProp.put(entry.getKey(), entry.getValue());
+      }
+      return nextProp;
+    } catch (net.oauth.OAuthException e) {
+      logger.warn("BasicLTIUtil.signProperties OAuth Exception {}", e.getMessage());
+      throw new Error(e);
+    } catch (java.io.IOException e) {
+      logger.warn("BasicLTIUtil.signProperties IO Exception {}", e.getMessage());
+      throw new Error(e);
+    } catch (java.net.URISyntaxException e) {
+      logger.warn("BasicLTIUtil.signProperties URI Syntax Exception {}", e.getMessage());
+      throw new Error(e);
+    }
+  }
+
+
+
+  /**
+  * Create the HTML to render a POST form and then automatically submit it.
+  *
+  * @param cleanProperties
+  * @param endpoint
+  *  The LTI launch url.
+  * @param launchtext
+  *  The LTI launch text. Used if javascript is turned off.
+  * @param debug
+  *  Useful for viewing the HTML before posting to end point.
+  * @param extra
+  *  Useful for viewing the HTML before posting to end point.
+  * @return the HTML ready for IFRAME src = inclusion.
+  */
+  public static String postLaunchHTML(
+      final Map<String, String> cleanProperties, String endpoint,
+      String launchtext, boolean debug, Map<String,String> extra) {
+    // Assume autosubmit is true for backwards compatibility
+    boolean autosubmit = true;
+    return postLaunchHTML(cleanProperties, endpoint, launchtext, autosubmit, debug, extra);
+  }
+
+
+  /**
+   * Create the HTML to render a POST form and then automatically submit it.
+   *
+   * @param cleanProperties
+   * @param endpoint
+   *  The LTI launch url.
+   * @param launchtext
+   *  The LTI launch text. Used if javascript is turned off.
+   * @param autosubmit
+   *  Whether or not we want the form autosubmitted
+   * @param extra
+   *  Useful for viewing the HTML before posting to end point.
+   * @return the HTML ready for IFRAME src = inclusion.
+   */
+  public static String postLaunchHTML(
+      final Map<String, String> cleanProperties, String endpoint,
+      String launchtext, boolean autosubmit, boolean debug,
+      Map<String,String> extra) {
+
+    if (cleanProperties == null || cleanProperties.isEmpty()) {
+      throw new IllegalArgumentException(
+          "cleanProperties == null || cleanProperties.isEmpty()");
+    }
+    if (endpoint == null) {
+      throw new IllegalArgumentException("endpoint == null");
+    }
+    Map<String, String> newMap = null;
+    if (debug) {
+      // sort the properties for readability
+      newMap = new TreeMap<String, String>(cleanProperties);
+    } else {
+      newMap = cleanProperties;
+    }
+    StringBuilder text = new StringBuilder();
+    // paint form
+    String submitUuid = UUID.randomUUID().toString().replace("-","_");
+    text.append("<div id=\"ltiLaunchFormArea_");
```

Comment:
This whole block is hard to read and seems like it should be a template. I'm also not sure if and how user input is passed in here and if this is safe. E.g. thinking of passing a backspace character in to manipulate the web page.